src/server/string_family.cc

```diff
@@ -448,9 +450,15 @@ SinkReplyBuilder::MGetResponse OpMGet(util::fb2::BlockingCounter wait_bc, bool f
         memcpy(next, v.data(), v.size());
         wait_bc->Dec();
       };
+      db_slice.IncRamMisses();
       shard->tiered_storage()->Read(t->GetDbIndex(), it.key(), it->second, std::move(cb));
     } else {
+      db_slice.IncRamHits();
       CopyValueToBuffer(it->second, next);
+      it->first.SetTouched(true);
+      if (it->second.HasStashPending()) {
+        shard->tiered_storage()->CancelStash(index, it.key(), &it->second);
+      }
```

Comment:
Same here, all that code just makes high level code more difficult to read 😢

Reply:
And you also won't need to expose changing flags on a const CompactObject

Reply:
Thanks, I will check again and will add comments here